framework/src/main/java/org/checkerframework/framework/ajava/JointJavacJavaParserVisitor.java

```diff
@@ -0,0 +1,2382 @@
+                    && isNoArgumentConstructor(javacMember)
+                    && (!hasNextJavaParser || !isNoArgumentConstructor(javaParserMember))) {
+                hasNextJavac = javacIter.hasNext();
+                javacMember = hasNextJavac ? javacIter.next() : null;
+                continue;
+            }
+
+            // In javac, a line like int i = 0, j = 0 is expanded as two sibling VariableTree
+            // instances. In JavaParser this is one FieldDeclaration with two nested
+            // VariableDeclarators. Match the declarators with the VariableTrees.
+            if (hasNextJavaParser && javaParserMember.isFieldDeclaration()) {
+                for (VariableDeclarator decl :
+                        javaParserMember.asFieldDeclaration().getVariables()) {
+                    assert hasNextJavac;
+                    assert javacMember.getKind() == Kind.VARIABLE;
+                    javacMember.accept(this, decl);
+                    hasNextJavac = javacIter.hasNext();
+                    javacMember = hasNextJavac ? javacIter.next() : null;
+                }
+
+                hasNextJavaParser = javaParserIter.hasNext();
+                javaParserMember = hasNextJavaParser ? javaParserIter.next() : null;
+                continue;
+            }
+
+            assert hasNextJavac;
+            assert hasNextJavaParser;
+            javacMember.accept(this, javaParserMember);
+
+            hasNextJavac = javacIter.hasNext();
+            javacMember = hasNextJavac ? javacIter.next() : null;
+
+            hasNextJavaParser = javaParserIter.hasNext();
+            javaParserMember = hasNextJavaParser ? javaParserIter.next() : null;
+        }
+
+        assert !hasNextJavac;
+        assert !hasNextJavaParser;
+    }
+
+    /**
+     * Visits the the members of an anonymous class body.
+     *
+     * <p>In normal classes, javac inserts a synthetic no-argument constructor if no constructor is
+     * explicitly defined, which is skipped when visiting members. Anonymous class bodies may
+     * introduce constructors that take arguments if the constructor invocation that created them
+     * was passed arguments. For example, if {@code MyClass} has a constructor taking a single
+     * integer argument, then writing {@code new MyClass(5) { }} expands to the javac tree
+     *
+     * <pre>{@code
+     * new MyClass(5) {
+     *     (int arg) {
+     *         super(arg);
+     *     }
+     * }
+     * }</pre>
+     *
+     * <p>This method skips these synthetic constructors.
+     *
+     * @param javacBody body of an anonymous class body
+     * @param javaParserMembers list of members for the anonymous class body of an {@code
+     *     ObjectCreationExpr}
+     */
+    public void visitAnonymouClassBody(
+            ClassTree javacBody, List<BodyDeclaration<?>> javaParserMembers) {
+        List<Tree> members = new ArrayList<>(javacBody.getMembers());
+        while (!members.isEmpty()) {
```

Comment:
Why is this a `while` loop rather than an `if` test?  I think there can be at most one initializer to remove.